tests/test_diaPipe.py

```diff
@@ -114,21 +114,31 @@ def _testRun(self, doPackageAlerts=False, doSolarSystemAssociation=False):
         # execution in the test environment.
         def concatMock(data):
             return MagicMock(spec=pd.DataFrame)
+
+        def metadataMock(self, nUpdatedDiaObjects,
+                         nUnassociatedDiaObjects,
+                         nNewDiaObjects,
+                         nTotalSsObject,
+                         nAssociatedSsObjects):
+            pass
+
         with patch.multiple(
             task, **{task: DEFAULT for task in subtasksToMock + ["apdb"]}
         ):
             with patch('lsst.ap.association.diaPipe.pd.concat',
                        new=concatMock):
-                result = task.run(diaSrc,
-                                  ssObjects,
-                                  diffIm,
-                                  exposure,
-                                  template,
-                                  ccdExposureIdBits,
-                                  "g")
-                for subtaskName in subtasksToMock:
-                    getattr(task, subtaskName).run.assert_called_once()
-                pipeBase.testUtils.assertValidOutput(task, result)
+                with patch('lsst.ap.association.diaPipe.DiaPipelineTask._add_association_meta_data',
+                           new=metadataMock):
```

Comment:
I guess I'd prefer that the subtask mocks actually return numbers that could be inserted into the metadata.  Is mocking things this way overly constraining the interface of the internal function, requiring changes here if additional items are added?

Reply:
Okay. I'll try to work out how the mocks work. This was the quick fix because I didn't want to disentangle all the mocking.

Reply:
That isn't necessarily your job; you can merge this and file a ticket for the package author(s) to fix this.